vendor/bat-native-ads/src/bat/ads/history_info.cc

```diff
@@ -19,46 +21,48 @@ HistoryInfo& HistoryInfo::operator=(const HistoryInfo& info) = default;
 
 HistoryInfo::~HistoryInfo() = default;
 
-std::string HistoryInfo::ToJson() const {
-  std::string json;
-  SaveToJson(*this, &json);
-  return json;
-}
+base::Value::Dict HistoryInfo::ToValue() const {
+  base::Value::List history;
+  for (const auto& item : items) {
+    history.Append(item.ToValue());
+  }
 
-bool HistoryInfo::FromJson(const std::string& json) {
-  rapidjson::Document document;
-  document.Parse(json.c_str());
+  base::Value::Dict dict;
+  dict.Set("history", std::move(history));
+  return dict;
+}
 
-  if (document.HasParseError()) {
-    BLOG(1, helper::JSON::GetLastError(&document));
-    return false;
-  }
+bool HistoryInfo::FromValue(const base::Value::Dict& root) {
+  if (const auto* value = root.FindList("history")) {
+    for (const auto& item : *value) {
+      if (!item.is_dict())
+        continue;
 
-  if (document.HasMember("history")) {
-    for (const auto& item : document["history"].GetArray()) {
       HistoryItemInfo history_item;
-      rapidjson::StringBuffer buffer;
-      rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
-      if (item.Accept(writer) && history_item.FromJson(buffer.GetString())) {
-        items.push_back(history_item);
-      }
+      history_item.FromValue(item.GetDict());
+      items.push_back(history_item);
     }
   }
 
   return true;
 }
 
-void SaveToJson(JsonWriter* writer, const HistoryInfo& info) {
-  writer->StartObject();
+std::string HistoryInfo::ToJson() const {
+  std::string json;
+  CHECK(base::JSONWriter::Write(ToValue(), &json));
+  return json;
+}
+
+bool HistoryInfo::FromJson(const std::string& json) {
+  auto document =
```

Comment:
Ditto

Reply:
Fixed in 32eb3390638af962d9af9b6b989e2488f2151c10.